src/catapult/catapult.ml

```diff
@@ -137,27 +153,276 @@ let printf t format_string =
   let c = next_leading_char t in
   Printf.ksprintf t.print ("%c" ^^ format_string ^^ "\n") c
 
-let pp_time f =
-  let n = int_of_float @@ (f *. 1_000_000.) in
-  Printf.sprintf "%d" n
+module Event = struct
+  [@@@ocaml.warning "-37"]
 
-type event = int * string
+  type common =
```

Comment:
I don't understand the need for encoding all the stuff that we're never going to use. There may be bugs in this module which we'll probably never know about.  I don't know what `Process_sort_index` means and I hope I never will.
I'm scrolling past most of this code, but honestly I'd much rather we didn't have it.